Updated to angular-cli v1.0.0 and angular v4.0.0

coder925 opened the pull request:
Updated to angular-cli v1.0.0-beta.28.3 and angular/material v2.0.0-beta.1.

Steps taken:
```bash
npm install --save-dev angular-cli@latest
npm install
ng update
```
During the update the following file actions were taken:

- skip: README.md
- skip: src\app\app.component.css
- skip: src\app\app.component.html
- skip: src\app\app.component.ts
- skip: src\app\app.module.ts
- overwrite: src\environments\environment.prod.ts
- overwrite: src\environments\environment.ts
- skip: src\index.html
- overwrite: src\main.ts
- overwrite: src\polyfills.ts
- skip: src\styles.css
- overwrite: src\test.ts
- overwrite: src\tsconfig.json
- merge: angular-cli.json
- overwrite: .gitignore
- overwrite: karma.conf.js
- overwrite: package.json
- overwrite: protractor.conf.js
- overwrite: tslint.json
- added: .editorconfig
- added: e2e\app.e2e-spec.ts
- added: e2e\app.po.ts
- added: e2e\tsconfig.json
- added: src\app\app.component.spec.ts
- deleted: src\environments\environment.dev.ts

Created missing folder after update
`mkdir src\assets`

Updated Material according to documentation:
```
npm install --save @angular/material
npm install --save hammerjs 
npm install --save-dev @types/hammerjs
```

Updated firebase-tools:
`npm install --save-dev firebase-tools
`

Renamed classes to align with file naming and allow for easier angular-cli updates in the future.
renamed: MaterialAppModule > AppModule
renamed: Material2AppAppComponent > AppComponent

coder925 committed:
Updated to angular-cli v1.0.0-beta.28.3.

bogste commented:
Happy to see that you took the care to do it!
I have already done this and was planning a PR.

BTW, how did you get the AoT to work? It looks broken on my side?

coder925 commented:
I am glad you appreciate it. Actually, I did not do anything at all regarding AOT. Since I let angular-cli `ng update`overwrite both _main.ts_ and _tsconfig.json_ some previous AOT related code/config was stripped.

I assumed that the app worked since I get no error messages after running `ng serve --aot`. Please let me know if I have overlooked something. Thanks!

michael-bergman committed:
Upgraded to angular 4.0.0 and angular-cli 1.0.0.

coder925 commented:
Updated to angular-cli v1.0.0, angular v4.0.0 and angular/material v2.0.0-beta.2.

Steps taken:
1. Created a new app using angular-cli.  
    ```
    ng new M2App
    npm install --save @angular/material
    npm install --save @angular/animations
    npm install --save hammerjs
    npm install --save-dev firebase-tools
    ```
2. Merged the old directory with the new.
3. Fixed some deprecated syntax in _app.component.html_ and _material2-app-theme.scss_.

asiftasleem commented:
Please merge this PR or upgrade to latest versions.
